src/components/SignUpForm/index.jsx

```diff
@@ -1,5 +1,150 @@
 import React from "react";
+import { useForm } from "react-hook-form";
+import Joi from "joi";
+import { joiResolver } from "@hookform/resolvers/joi";
+import InputErrorMessage from "../InputErrorMessage";
+import "./index.scss";
 
-export default function SignUpForm() {
-  return <div>SignUpForm Component</div>;
+const schema = Joi.object({
+  firstName: Joi.string().required().messages({
+    "string.empty": `Please provide your first name`,
+  }),
+  lastName: Joi.string().required().messages({
+    "string.empty": `Please provide your last name`,
+  }),
+  email: Joi.string()
+    .required()
+    .email({ tlds: { allow: ["com", "net", "edu"] } })
```

Comment:
delete the tlds. no reason to restrict what emails can register

Reply:
I put just for learning purpose :)
I will remove it